exonum-java-binding/CHANGELOG.md

```diff
@@ -15,28 +15,43 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+<!-- TODO: Set release date -->
+## [0.4] - 2019-02-DD
+
+This release is based on the latest Exonum version, 0.10, 
+and brings the features and improvements outlined below.
+If you are upgrading an existing Java service, consult the migration guide.
+<!-- TODO: Reference it -->
+
 ### Added
-- Support of new transaction message format added in Exonum 0.10. (#534)
-- Support of core schema API. (#548, #549, #571, #573)
+- Support of new transaction message format added in Exonum 0.10. It allows service 
+  developers to use any serialization format, and the framework to perform signature verification
+  automatically. Both the service and client APIs are migrated. (#534, #574)
+- `com.exonum.binding.blockchain.Blockchain`, allowing services to read data, 
+  maintained by the framework (blocks, transaction messages, transaction execution results, etc.).
+  (#548, #549, #571, #573)
 - Support of `Service#afterCommit(BlockCommittedEvent event)` method
   that is invoked after each block commit event. (#550)
-- Support of Json serialization in a common way. (#611)
-- `--jvm-debug` command line argument that allows JDWP debugging of node. (#629)
-- Support of separated messages format Exonum v0.10. (#574)
+- `com.exonum.binding.common.serialization.json.JsonSerializer` providing JSON serialization 
+  support of any objects, including Exonum built-in types. (#611)
+- Support of passing arbitrary arguments to the JVM when the node is launched via
+  `--jvm-args-prepend` and `--jvm-args-append` CLI flags. (#629)
+- `--jvm-debug` application command line argument that allows JDWP debugging of a node. (#629)
 - `ListIndexProxy#stream` to enable stream processing of list elements. (#661)
+- Support of running services on Java 11 runtime.
```

Comment:
Nit: on Travis, we have the build job with Java 11 configured for Linux only (and it is allowed to fail AFAIR). But probably CHANGELOG has nothing to do with this.

Reply:
Yes, but we also run our internal system tests on 11.